### PR TITLE
Add Authorization endpoints to API which lists accounts.

### LINKED ISSUE #99: Provide list of accounts for an autorized session
Basically just need a list of accounts in a way that lets us switch between them for the purposes of persisting to that account.

`/api/v1/authorization/accounts` seems like the way to go.

needed for wrapping PRX/www.prx.org#210 up in a nice bow

### PR DESCRIPTION
**Pull request by chrisrhoden**
resolves #99

### CODE REVIEW

**Comment by chrisrhoden** [comment]
closes #99

File: app/controllers/api/authorizations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# encoding: utf-8
+
+class Api::AuthorizationsController < Api::BaseController
+  def resource
+    Authorization.new(User.find(prx_auth_token.user_id))
```
Comment (kookster) [inline app/controllers/api/authorizations_controller.rb]: I might DRY this out a bit and put `User.find(prx_auth_token.user_id)` into a find or get user method.

File: app/controllers/api/authorizations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def resources
+    Authorization.new(User.find(prx_auth_token.user_id)).accounts
+  end
+
+  def index_collection
```
Comment (kookster) [inline app/controllers/api/authorizations_controller.rb]: I am reconciling myself to this, but it seems odd to have accounts and authorization resources returned by the same controller.

I can't think of a better option really; maybe making a new controller just for this nested `accounts` index action - but that seems like it is not worth the effort.

File: app/models/base_model.rb [hunk 3]
```diff
@@ -4,18 +4,7 @@ class BaseModel < ActiveRecord::Base
...
-  attr_accessor :is_root_resource
-
-  def is_root_resource
-    !!@is_root_resource
-  end
-
-  def show_curies
-    is_root_resource
-  end
+  include RepresentedModel
```
Comment (kookster) [inline app/models/base_model.rb]: Nice idea, good refactoring.

**Comment by kookster** [comment]
@chrisrhoden I had only minor feedback on this, LGTM.
If you want to DRY up that User.find method, fine, if not, fine - I'll merge as is.

**Comment by chrisrhoden** [comment]
@kookster let me know if I missed something

**Pull request merged**